composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanner.kt

```diff
@@ -0,0 +1,52 @@
+package com.rahulrav.camera.scan
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.flow.Flow
+
+@Composable
+fun CameraScanner(
+    viewModel: CameraScanViewModel,
+    modifier: Modifier = Modifier
+) {
+    when (val state = viewModel.state.value) {
+        is ScanState.StartingScan -> Scanning(modifier)
+        is ScanState.ScanResults -> ShowResults(state.cameras, { viewModel.pair(it) }, modifier)
```

Comment:
The programmatic pairing requires a bond prior to scan. We can do better on Android though, given we have the ability to programmatically request a bond. I had this in a prior version of the app, but I got rid of this for the MPP version given iOS APIs don't really support programmatic bonding.

Reply:
Will tackle separately, need to investigate better (and with an iPhone)